### PR TITLE
add libva support

### PR DESCRIPTION
**Pull request by activebug202**
Signed-off-by: bozhengx <box.zheng@intel.com>

### CODE REVIEW

File: drmhwcomposer.h [hunk 1]
```diff
@@ -129,6 +140,8 @@ enum class DrmHwcBlending : int32_t {
 
 struct DrmHwcLayer {
   buffer_handle_t sf_handle = NULL;
+  buffer_handle_t sf_va_handle = NULL;
```
Comment (feijiang1) [inline drmhwcomposer.h]: don't need add

File: drmhwcomposer.h [hunk 2]
```diff
@@ -129,6 +140,8 @@ enum class DrmHwcBlending : int32_t {
...
+  std::map<uint32_t, DrmHwcLayer *, std::greater<int>> va_z_map;
```
Comment (feijiang1) [inline drmhwcomposer.h]: not easy to understand

File: drmhwcomposer.h [hunk 3]
```diff
@@ -137,17 +150,41 @@ struct DrmHwcLayer {
...
+  }
+  std::map<uint32_t, DrmHwcLayer *, std::greater<int>> getVaLayerMapData(){
+    return va_z_map;
+  }
+  void SetVaLayerData(buffer_handle_t handle){
+    sf_va_handle = handle;
+  }
+  buffer_handle_t get_valayer_handle() const {
+    return sf_va_handle;
+  }
```
Comment (feijiang1) [inline drmhwcomposer.h]: need delete

**Comment by YuanjunHuang** [comment]
What is this PR for? what is the difference of va layer with original video layer?

File: platformminigbm.cpp [hunk 4]
```diff
@@ -67,19 +91,62 @@ int DrmMinigbmImporter::Init() {
   return 0;
 }
 
-int DrmMinigbmImporter::ImportBuffer(buffer_handle_t handle, hwc_drm_bo_t *bo) {
+#ifdef ENABLE_DUMP_YUV_DATA
```
Comment (feijiang1) [inline platformminigbm.cpp]: need use property to enable yuv dump

File: platformminigbm.cpp [hunk 5]
```diff
@@ -67,19 +91,62 @@ int DrmMinigbmImporter::Init() {
...
+#define DUMP_DATA_FILE "/data/temp"
```
Comment (feijiang1) [inline platformminigbm.cpp]: need add width/height/format info in the dump file

File: drmhwctwo.cpp [hunk 6]
```diff
@@ -727,50 +746,82 @@ HWC2::Error DrmHwcTwo::HwcDisplay::SetVsyncEnabled(int32_t enabled) {
   return HWC2::Error::None;
 }
 
+//check whether the layer with the lease two zorder is the video layer
+bool DrmHwcTwo::HwcDisplay::WhetherVideoLayer(std::map< uint32_t,
```
Comment (Shao-Feng) [inline drmhwctwo.cpp]: Is better to name as "IsXXXX" or "ContainXXX" for the method which return bool

Comment (activebug202) [reply drmhwctwo.cpp]: > 
> 
> Is better to name as "IsXXXX" or "ContainXXX" for the method which return bool

Get it.I'll change.

**Comment by feijiang1** [comment]
two commits are submitted, confused, please submit a new one

**Pull request closed without merging**